src/Dapr.AspNetCore/DaprEndpointRouteBuilderExtensions.cs

```diff
@@ -57,14 +57,16 @@ private static IEndpointConventionBuilder CreateSubscribeEndPoint(IEndpointRoute
                 var dataSource = context.RequestServices.GetRequiredService<EndpointDataSource>();
                 var entries = dataSource.Endpoints
                     .OfType<RouteEndpoint>()
-                    .Where(e => e.Metadata.GetMetadata<ITopicMetadata>()?.Name != null)   // only endpoints which have  TopicAttribute with not null Name.
+                    .Where(e => e.Metadata.GetMetadata<ITopicMetadata>()?.Name != null)   // only endpoints which have TopicAttribute with not null Name.
                     .Distinct()
-                    .Select(e => (e.Metadata.GetMetadata<ITopicMetadata>().PubsubName, e.Metadata.GetMetadata<ITopicMetadata>().Name, e.Metadata.GetMetadata<IRawTopicMetadata>()?.EnableRawPayload, e.RoutePattern));
-
-
+                    .Select(e => (e.Metadata.GetMetadata<ITopicMetadata>().PubsubName, e.Metadata.GetMetadata<ITopicMetadata>().Name, e.Metadata.GetMetadata<IRawTopicMetadata>()?.EnableRawPayload, e.Metadata.GetMetadata<ITopicMetadata>().Match, e.Metadata.GetMetadata<ITopicMetadata>().Priority, e.RoutePattern))
+                    .GroupBy(e => new { e.PubsubName, e.Name })
+                    .Select(e => e.OrderBy(e => e.Priority))
+                    .SelectMany(e => e);
                 var logger = context.RequestServices.GetService<ILoggerFactory>().CreateLogger("DaprTopicSubscription");
 
                 List<Subscription> subscriptions = new();
+                Dictionary<string, Subscription> subscriptionMap = new();
```

Comment:
We shouldn't need both a list and dictionary. Just use a dictionary and you can access the values using `subscriptionMap.Values.OrderBy( ... )`. 

You want the explicit ordering for testability since dictionary does not guarantee iteration order.